01_Data/src/layers/sequelize/model/Tariff/TariffElement.ts

```diff
@@ -0,0 +1,175 @@
+import {PriceComponent} from './PriceComponent';
+import {TariffRestrictions} from './TariffRestrictions';
+import {BelongsTo, Column, DataType, ForeignKey, Model, Table,} from "sequelize-typescript";
+import {
+    assert,
+    DateOnly,
+    DayOfWeek,
+    DurationInSeconds,
+    hasDuplicates,
+    isTariffId,
+    Namespace,
+    notEmpty,
+    safelyMap,
+    TariffId,
+    Time
+} from "@citrineos/base";
+import {Tariff} from "./Tariffs";
+import {ReservationRestrictionType} from "./ReservationRestrictionType";
+import {TimeColumn} from "../../types/TimeColumn";
+import {DateOnlyColumn} from "../../types/DateOnlyColumn";
+
+@Table
+export class TariffElement extends Model implements TariffElementData {
+
+  static readonly MODEL_NAME: string = Namespace.TariffElement;
+
+  public static newInstance(data: TariffElementData): TariffElement {
+    return TariffElement.build({...data});
+  }
+
+  @BelongsTo(() => Tariff)
+  declare tariff: Tariff;
+
+  @ForeignKey(() => Tariff)
+  @Column({
+    type: DataType.STRING,
+    validate: {
+      isTariffId: (value: string) => assert(isTariffId(value), `Invalid tariffId: ${value}`),
+    }
+  })
+  declare tariffId: TariffId;
+
+  @Column({
+    type: DataType.JSON,
+    allowNull: false,
+    validate: {
+      notNull: true,
+      atLeastOne: (value: PriceComponent[]) => assert(notEmpty(value), `priceComponents cannot be empty`),
+      hasNoDuplicates: (value: PriceComponent[]) => assert(!hasDuplicates(value, 'type'), 'priceComponents cannot contain duplicates'),
+    }
+  })
+  declare priceComponents: PriceComponent[];
+
+  @Column(DataType.VIRTUAL)
```

Comment:
I think it's preferable to maintain tariff restrictions within the tariff elements table instead of creating a separate table for them. Individual restrictions are private within this class, and a virtual data type was used to present restrictions as a single object from the TariffElement.